File: semestre-1/algoritmos/MR-Alg-03/.py/MR-Alg-03-Ex-09.py
feriados = {
    "contraternização universal": (1, "janeiro"),
    "tiradentes": (21, "abril"),
    "dia do trabalho": (1, "maio"),
    "independência do brasil": (7, "setembro"),
    "nossa senhora aparecida": (12, "outubro"),
    "finados": (2, "novembro"),
    "proclamação da república": (15, "novembro"),
    "natal": (25, "dezembro")
}

meses = (
    "janeiro",
    "fevereiro",
    "março",
    "abril",
    "maio",
    "junho",
    "julho",
    "agosto",
    "setembro",
    "outubro",
    "novembro",
    "dezembro"
)

mes = input("Digite um mês: ")
dia = int(input("Digite um dia do mês: "))

# Verifica mes e dia válidos
if mes.lower() not in meses or dia < 1 or dia > 31:
    print("Mês ou dia inválidos!")
else:
    for feriado_nome, (feriado_dia, feriado_mes) in feriados.items():
        if feriado_dia == dia and feriado_mes.lower() == mes.lower():
            print(f"{dia} de {mes} é um feriado nacional: {feriado_nome.title()}")

File: semestre-1/algoritmos/MR-Alg-03/.c/MR-Alg-03-Ex-09.c
/*A tabela abaixo mostra os feriados nacionais brasileiros que caem sempre no
mesmo dia (em oposição aos feriados variáveis como carnaval e corpus christi).

Feriado                         Data
Confraternização universal      1o. de janeiro
Tiradentes                      21 de abril
Dia do trabalho                 1o. de maio
Independência do Brasil         7 de setembro
Nossa Senhora Aparecida         12 de outubro
Finados                         2 de novembro
Proclamação da República        15 de novembro
Natal                           25 de dezembro

Escreva um programa Python que leia do usuário o mês e o dia de uma determinada data. Se
o mês e o dia corresponderem a uma das datas da tabela acima, seu programa deve exibir o
nome do feriado. Caso contrário o programa deve informar que o dia e o mês informados não
correspondem a um feriado nacional.*/

#include <stdio.h>

void main() {
    int dia, mes;
    printf("Dia: ");
    scanf("%d", &dia);

    printf("Numero do mes: ");
    scanf("%d", &mes);

    int feriados_data[][2] = {
        {1, 1},
        {21, 4},
        {1, 5},
        {7, 9},
        {12, 10},
        {2, 11},
        {15, 11},
        {25, 12}
    };

    char *feriados_nome[] = {
        "Confraternizacao universal",
        "Tiradentes",
        "Dia do trabalho",
        "Independencia do Brasil",
        "Nossa Senhora Aparecida",
        "Finados",
        "Proclamacao da Republica",
        "Natal"
    };

    size_t num_feridos = sizeof feriados_data / sizeof *feriados_data;
    for (int i = 0; i < num_feridos; i++) {
        if (feriados_data[i][0] == dia && feriados_data[i][1] == mes) {
            printf("%c%d/%c%d: %s\n", dia < 10 ? '0' : '\0', dia, mes < 10 ? '0' : '\0', mes, feriados_nome[i]);
        }
    }
}
